src/Microsoft.VisualStudio.LanguageServices.Razor/DefaultTagHelperCompletionService.cs

```diff
@@ -0,0 +1,153 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Linq;
+using Microsoft.AspNetCore.Razor.Language;
+
+namespace Microsoft.VisualStudio.LanguageServices.Razor
+{
+    [Export(typeof(TagHelperCompletionService))]
+    internal class DefaultTagHelperCompletionService : TagHelperCompletionService
+    {
+        [Import(typeof(TagHelperFactsService))]
+        public TagHelperFactsService TagHelperFactsService { get; set; }
```

Comment:
Use `ImportingConstructor` here - we don't really need to use property injection, and that will be more natural for tests.